web/cypress/e2e/auto-tax-filings.spec.ts

```diff
@@ -0,0 +1,101 @@
+
+    completeBusinessStructureTask({ legalStructureId });
+
+    onDashboardPage.getEditProfileLink().should("exist");
+    cy.get('[data-task="apply-for-shop-license"]').first().click();
+    cy.get('button[data-testid="cta-secondary"]').first().click();
+    cy.intercept(`${Cypress.env("API_BASE_URL")}/api/users/*`).as("userAPI");
+
+    cy.visit("/profile");
+    cy.get('input[data-testid="businessName"]').type(businessName);
+    cy.get('button[data-testid="save"]').first().click();
+    openFormationDateModal();
+    selectDate("04/2021");
+    selectLocation("Allendale");
+    clickModalSaveButton();
+
+    cy.get('[data-testid="register-for-taxes"]').first().click();
+    cy.get('input[name="taxId"]').type("123456789098");
+    cy.get("button").contains("Save").click();
+    cy.get(`[data-testid="back-to-dashboard"]`).first().click({ force: true });
+    cy.get('[data-testid="cta-funding-nudge"]').first().click();
+    cy.get('[data-testid="get-tax-access"]').should("not.exist");
+    cy.get('[data-testid="alert-content-container"]').should("exist");
+  });
+
+  it("does not complete auto tax filing if missing business name", () => {
+    const industry = LookupIndustryById("cosmetology");
+    const legalStructureId = "limited-liability-company";
+
+    completeNewBusinessOnboarding({ industry });
+
+    completeBusinessStructureTask({ legalStructureId });
+
+    onDashboardPage.getEditProfileLink().should("exist");
+    cy.get('[data-task="apply-for-shop-license"]').first().click();
+    cy.get('button[data-testid="cta-secondary"]').first().click();
+    cy.intercept(`${Cypress.env("API_BASE_URL")}/api/users/*`).as("userAPI");
+    cy.get(`[data-testid="back-to-dashboard"]`).first().click({ force: true });
+    openFormationDateModal();
+    selectDate("04/2021");
+    selectLocation("Allendale");
+    clickModalSaveButton();
+
+    cy.get('[data-testid="register-for-taxes"]').first().click();
+    cy.get('input[name="taxId"]').type("123456789098");
+    cy.get("button").contains("Save").click();
+    cy.get(`[data-testid="back-to-dashboard"]`).first().click({ force: true });
+    cy.get('[data-testid="cta-funding-nudge"]').first().click();
+    cy.get('[data-testid="get-tax-access"]').should("exist");
+    cy.get('[data-testid="alert-content-container"]').should("not.exist");
+  });
+
+  it.only("does not complete auto tax filing if missing tax Id", () => {
+    const industry = LookupIndustryById("cosmetology");
+    const legalStructureId = "limited-liability-company";
+    const businessName = "Make Me Pretty";
+
+    completeNewBusinessOnboarding({ industry });
+
+    completeBusinessStructureTask({ legalStructureId });
+
+    onDashboardPage.getEditProfileLink().should("exist");
+    cy.get('[data-task="apply-for-shop-license"]').first().click();
+    cy.get('button[data-testid="cta-secondary"]').first().click();
+    cy.intercept(`${Cypress.env("API_BASE_URL")}/api/users/*`).as("userAPI");
```

Comment:
dust - so it looks like you're getting a license in to have something actually appear in the calendar when it's rendered rather than an empty calendar?

If thats right, is that necessary? Like would it not be slightly simpler to check for the tax calendar itself not existing or being pending and then check for it as an empty tax calendar? (Or is there some weird logic here that prevents that from being the case)?